doc: Cleanup/org Distinguishing Features list

gregshue opened the pull request:
Clean up the organization of the Distinguishing Features list re-grouped under:
- Reusable, extensible framework
- Configurable, extensible domain architecture
- Extensible reusable asset base

Verified by:

1. make html-fast

Fix #58201

nashif commented:
now sure how you made the changes, but it is very difficult to ready through this and see what have you changed, i.e. the diff is one single block

legrand-gregshue committed:
doc: Reorg Dist Features - new group headings

Clean up the organization of the Distinguishing Features list
re-grouped under:
- Reusable, extensible framework
- Configurable, extensible domain architecture
- Extensible reusable asset base

This organization better reflects the types of
reusability and scopes of the distinguishing features.

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>

legrand-gregshue committed:
doc: Reorg Dist Features - Modular, CPU archs, rsrc defns, host OS

Clean up the organization of the Distinguishing Features list.

Relocated/reformatted:
- Highly configurable / Modular for flexibility
- Cross (CPU) Architecture
- Compile-time resource definition
- Native Linux, macOS, and Windows Development

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>

legrand-gregshue committed:
doc: Reorg Dist Features - Kernel, Sched, MP

Clean up the organization of the Distinguishing Features list.

Relocated/reformatted:
- Extensive suite of Kernel services
- Multiple Scheduling Algorithms
- Memory Protection

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>

legrand-gregshue committed:
doc: Reorg Dist Features - Driver model, Ntwk, BLE 5.0

Clean up the organization of the Distinguishing Features list.

Relocated/reformatted:
- Optimized Device Driver Model
- Native Networking Stack supporting multiple protocols
- Bluetooth Low Energy 5.0 support

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>

legrand-gregshue committed:
doc: Reorg Dist Features - VFS, log, shell, settings, NVS

Clean up the organization of the Distinguishing Features list.

Relocated/reformatted:
- Virtual File System Interface with LittleFS and FATFS Support
- Power multi-backend logging Framework
- User friendly and full-featured Shell interface
- Settings on non-volatile storage
- Non-volatile storage (NVS)

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>

legrand-gregshue committed:
doc: Reorg Dist Features - Devicetree, Native POSIX port

Clean up the organization of the Distinguishing Features list.

Relocated/reformatted:
- Devicetree Support
- Natie POSIX port

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>

legrand-gregshue committed:
doc: Add Dist Features - Kconfig, twister, extensible

Extend the Distinguishing Features list for framework.

- Project-specific feature-based configuration management
- Test framework, test runner, and code coverage tools
- Extensible, modulear workspace
- Extensible documentation
- Generated linker directives and system interfaces
- Extensible build system

Verified by:

1. make html-fast

Signed-off-by: Gregory Shue <gregory.shue@legrand.com>
